### PR TITLE
Adding middleware functions, replacing deprecated grpc.Dial call

### PR DESCRIPTION
**Pull request by sre-is-josebiro**

### CODE REVIEW

**Comment by sre-is-abrahms** [comment]
Relevant:  https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md#dialing-in-grpc

File: handlers/middleware.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+// NewHttpHandler wraps the passed handler and enriches it with a guard.
+func NewHttpHandler(handler http.Handler, gn string, fn *string, pb *int32, dw *float32, kv *map[string]string) http.Handler {
+	return NewMiddleware(gn, fn, pb, dw, kv)(handler)
+}
+
+// NewMiddleware returns a Guard middleware
+// The handler returned by the middleware wraps a handler and provides a Guard
+// for each request
+func NewMiddleware(gn string, fn *string, pb *int32, dw *float32, kv *map[string]string) func(http.Handler) http.Handler {
```
Comment (sre-is-abrahms) [inline handlers/middleware.go:24]: These variable names leave a lot to be desired. Mind changing them to be the same as the keys of the middleware below? They will be visible to users of the SDK, as I understand it.

**Review by sre-is-abrahms — APPROVED** [review]

File: handlers/middleware.go (hunk 1 above)
Comment (sre-is-josebiro) [reply handlers/middleware.go:24]: Yeah, good call.

**Comment by sre-is-josebiro** [comment]
> Relevant: https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md#dialing-in-grpc

We're not using any of the deprecated Dial options from what I can see, so this should be as simple as changing the function call.

**Review by sre-is-nathan — APPROVED** [review]

**Pull request merged**